gpu-alloc/src/allocator.rs

```diff
@@ -397,6 +397,71 @@ where
         Err(AllocationError::OutOfDeviceMemory)
     }
 
+    /// Creates a memory block from an existing memory allocation.
+    ///
+    /// This function allows the [`GpuAllocator`] to manage memory allocated outside of the typical
+    /// [`GpuAllocator::alloc`] family of functions.
+    ///
+    /// # Usage
+    ///
+    /// If you need to import external memory, such as a Win32 `HANDLE` or a Linux `dmabuf`, import the device
+    /// memory using the graphics api and platform dependent functions. Once that is done, call this function
+    /// to make the [`GpuAllocator`] responsible for managing the imported memory as a dedicated [`MemoryBlock`].
+    ///
+    /// When calling this function, you **must** ensure there are [enough remaining allocations](GpuAllocator::remaining_allocations).
+    ///
+    /// # Safety
+    ///
+    /// - The `memory` must be allocated with the same device that was provided to create this [`GpuAllocator`]
+    ///   instance.
+    /// - The `memory` must be valid.
+    /// - The `memory_type`, `props`, `offset` and `size` must match the memory type, properties offset and
+    ///   size of the memory allocation.
+    /// - There must be enough remaining allocations.
+    /// - The memory allocation must not come from an existing memory block created by this allocator.
+    /// - The underlying memory object must be deallocated using the returned [`MemoryBlock`] with
+    ///   [`GpuAllocator::dealloc`].
+    pub unsafe fn import_memory(
+        &mut self,
+        memory: M,
+        memory_type: u32,
+        props: MemoryPropertyFlags,
+        offset: u64,
+        size: u64,
+    ) -> MemoryBlock<M> {
+        // TODO: Before merge:
+        // - Should this memory belong to a heap?
```

Comment:
If it was allocated from a heap

Reply:
What type of memory would be classified as belonging to a heap?

Reply:
When memory object is allocated, it is allocated with a memory type, bounded to a memory heap.
When importing memory object, if it is associated with a heap, heap usage should be updated.